analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantCast.cs

```diff
@@ -18,161 +18,150 @@
-            invocation.GetMethodCallIdentifier() is { } methodName
-            && CastIEnumerableMethods.Contains(methodName.ValueText)
-            && semanticModel.GetSymbolInfo(invocation).Symbol is IMethodSymbol methodSymbol
-            && methodSymbol.IsExtensionOn(KnownType.System_Collections_IEnumerable)
-                ? methodSymbol
-                : null;
+    private static void ReportIssue(SonarSyntaxNodeReportingContext context, ExpressionSyntax expression, ITypeSymbol castType, Location location) =>
+        context.ReportIssue(Diagnostic.Create(Rule, location, castType.ToMinimalDisplayString(context.SemanticModel, expression.SpanStart)));
 
-        private static ITypeSymbol GetGenericTypeArgument(ITypeSymbol type) =>
-            type is INamedTypeSymbol returnType && returnType.Is(KnownType.System_Collections_Generic_IEnumerable_T)
-                ? returnType.TypeArguments.Single()
-                : null;
+    /// If the invocation one of the <see cref="CastIEnumerableMethods"/> extensions, returns the method symbol.
+    private static IMethodSymbol GetEnumerableExtensionSymbol(InvocationExpressionSyntax invocation, SemanticModel semanticModel) =>
+        invocation.GetMethodCallIdentifier() is { } methodName
+        && CastIEnumerableMethods.Contains(methodName.ValueText)
+        && semanticModel.GetSymbolInfo(invocation).Symbol is IMethodSymbol methodSymbol
+        && methodSymbol.IsExtensionOn(KnownType.System_Collections_IEnumerable)
+            ? methodSymbol
+            : null;
 
-        private static bool CanHaveNullValue(ITypeSymbol type) => type.IsReferenceType || type.Name == "Nullable";
+    private static ITypeSymbol GetGenericTypeArgument(ITypeSymbol type) =>
+        type is INamedTypeSymbol returnType && returnType.Is(KnownType.System_Collections_Generic_IEnumerable_T)
+            ? returnType.TypeArguments.Single()
+            : null;
 
-        private static Location GetReportLocation(InvocationExpressionSyntax invocation, bool methodCalledAsStatic)
-        {
-            if (!(invocation.Expression is MemberAccessExpressionSyntax memberAccess))
-            {
-                return invocation.Expression.GetLocation();
-            }
+    private static bool CanHaveNullValue(ITypeSymbol type) =>
+        type.IsReferenceType || type.Name == "Nullable";
 
-            return methodCalledAsStatic
+    private static Location GetReportLocation(InvocationExpressionSyntax invocation, bool methodCalledAsStatic) =>
+        invocation.Expression switch
+        {
+            MemberAccessExpressionSyntax memberAccess => methodCalledAsStatic
                 ? memberAccess.GetLocation()
-                : memberAccess.OperatorToken.CreateLocation(invocation);
-        }
+                : memberAccess.OperatorToken.CreateLocation(invocation),
+            _ => invocation.Expression.GetLocation()
```

Comment:
This case is not covered. You may be able to cover the case by using other access kinds: [sharplab](https://sharplab.io/#v2:EYLgtghglgdgNAFxBAzguATEBqAPgAQCYAGAWACgiBGC/YgAnyoDoAZWARwG5aGmA2Ri3YwOzAKIwArmACmAJwjAANrJ7laVAJwAKAESAeDcAg+3oCU6gG4R59a4oCe9ALz0YsgO71YCANoBdAG8AX3Vvf39beQcAeQAzAEEoiEcXN08w4j9/dQoAfWdIhwB+ZgBhVAQAHm8APh1zegB6RvoAWVkwYAUAIVgMWABzPIK7ZLjEhx9MsorqmAQ6hub6cVU5efiAY03ZFBRhl1H7caT7Iqm/GbQ5hfquJpbVjtl53ph+mAH6HQB7AGtkq4fgh6PJZBBlMpHD8YPRymg4PRgFIQSgfnJ3AALBSyeiyZQoWSmA7fAAiEAQsgAKlA5MwAHI/dzMckpFxUehFQrJACE9BArg8XnmF1MVyqtTuD3olPkMGs9iAA=)